src/HDF5.jl

```diff
@@ -1712,20 +1712,120 @@ function create_external_dataset(parent::Union{File,Group}, name::AbstractString
     create_dataset(parent, name, datatype(t), dataspace(sz); dcpl=dcpl)
 end
 
+"""
+    do_write_chunk(dataset::Dataset, offset, chunk_bytes::Vector{UInt8}, filter_mask=0)
+
+    Write a raw chunk at a given offset.
+    offset is 1-based of rank `ndims(dataset)` and must fall on a chunk boundary.
+"""
 function do_write_chunk(dataset::Dataset, offset, chunk_bytes::Vector{UInt8}, filter_mask=0)
```

Comment:
Are `do_write_chunk` and `do_read_chunk` really necessary methods? Couldn't they be implemented as the bodies of the corresponding `getindex` and `setindex!` methods? I assume deprecated bindings could easily be made for these.

Reply:
`do_write_chunk` precedes this pull request:
https://github.com/JuliaIO/HDF5.jl/blob/0fb17b0ef9f3eeeed7184dd691a00db959da8c1a/src/HDF5.jl#L1715-L1719

We could remove them, but I think it might be better to do that in another pull request.